File: codeforces/low_difficulty/difficulty_1000/4.chat_room_[58A]/ChatRoom.java
/* Author: José Rodolfo (jric2002) */
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.IOException;
public class ChatRoom {
  public static void main(String[] args) throws IOException {
    Character[] word = new Character[]{'h', 'e', 'l', 'l', 'o'};
    String s;
    Integer size_s, size_w, p;
    BufferedReader input = new BufferedReader(new InputStreamReader(System.in));
    s = input.readLine();
    size_s = s.length();
    size_w = word.length;
    p = 0;
    for (Integer i = 0; i < size_s; i++) {
      if (p < size_w) {
        if (s.charAt(i) == word[p]) {
          p++;
        }
      }
    }
    System.out.println((p == size_w) ? "YES" : "NO");
  }
}

File: codeforces/low_difficulty/difficulty_1000/4.chat_room_[58A]/chat_room.py
# Author: José Rodolfo (jric2002)
word = ['h', 'e', 'l', 'l', 'o']
s = str(input())
size_s = len(s)
size_w = len(word)
p = 0
for i in range(0, size_s):
  if (p < size_w):
    if (s[i] == word[p]):
      p += 1
print("YES") if (p == size_w) else print("NO")